fix: spells and runes interactions

kaleohanopahala opened the pull request:
- It is now possible to throw runes over stairs and holes (Global Condition), and use spells on walls. 
- An efficiency improvement was also made to avoid unnecessary reallocations.
- Correct actions like a global tibia.

![image](https://github.com/user-attachments/assets/b3978419-674b-488a-984b-ac5549ea4d77)
![image](https://github.com/user-attachments/assets/d9ff4c66-e773-4ddd-a460-f171d2beb2ca)
![image](https://github.com/user-attachments/assets/8ecb870e-f8d1-4879-b887-d98ae7d118aa)
![image](https://github.com/user-attachments/assets/405828fa-d79d-4475-812b-9e9526d620f8)


Maybe can close issue #2446 and #2794


#Reopened because I accidentally deleted the repository.

kaleohanopahala committed:
Update combat.cpp

kaleohanopahala committed:
Update combat.hpp

github-actions[bot] committed:
Code format - (Clang-format)

kaleohanopahala committed:
Update spells.cpp

kaleohanopahala committed:
fix sonar warning

majestyotbr committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Fix XmasGame issue report.

kaleohanopahala committed:
Fix magic wall rune

github-actions[bot] committed:
Lua code format - (Stylua)

kaleohanopahala committed:
fix  wild growth rune

github-actions[bot] committed:
Lua code format - (Stylua)

kaleohanopahala committed:
fix a mistake

oops.

kaleohanopahala committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Update magic_wall.lua

fix conflits.

kaleohanopahala committed:
Update wild_growth.lua

fix conflits

kaleohanopahala committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Merge branch 'main' into spells

murilo09 committed:
Merge branch 'main' into spells

murilo09 commented:
It's not yet possible to throw an avalanche at a magic wall (ID 2128 and 10181).
https://youtu.be/Q5ItUPYRZv4?t=510

kaleohanopahala committed:
Fix Throw on magic walls.

Fix murilo09 issue.

kaleohanopahala commented:
> It's not yet possible to throw an avalanche at a magic wall (ID 2128 and 10181). https://youtu.be/Q5ItUPYRZv4?t=510

Can you test again?
Changes on combat.cpp.

kaleohanopahala committed:
fix condition check

Now the aggressive flag check is the first condition to be evaluated. If aggressive is false, the function immediately returns RETURNVALUE_NOERROR, bypassing all the other checks that follow.

kaleohanopahala committed:
Merge branch 'main' into spells

murilo09 commented on src/creatures/combat/combat.cpp:
```diff
@@ -252,19 +252,33 @@ ReturnValue Combat::canTargetCreature(const std::shared_ptr<Player> &player, con
+
+		if (!Cannotthrow) {
+			return RETURNVALUE_CANNOTTHROW;
+		}
```
```suggestion
		bool canThrow = false;

		if (const auto fieldList = tile->getItemList()) {
			for (const auto &findfield : *fieldList) {
				if (findfield && (findfield->getID() == ITEM_MAGICWALL || findfield->getID() == ITEM_MAGICWALL_SAFE)) {
					canThrow = true;
					break;
				}
			}
		}

		if (!canThrow) {
			return RETURNVALUE_CANNOTTHROW;
		}
```

kaleohanopahala committed:
fix

Co-authored-by: murilo09 <78226931+murilo09@users.noreply.github.com>

murilo09 commented:
I’ll test it tomorrow

kaleohanopahala committed:
Merge branch 'main' into spells

kaleohanopahala committed:
Merge branch 'main' into spells

W4gNII commented:
hello @kaleohanopahala I have a problem when you cast a spell on a wall in the south position 0 sqm's away

note: only occurred on walls in the south position, spells and runes ignore a

https://github.com/user-attachments/assets/4d698056-f559-42a3-9f43-3c0f100dc3a2

 wall mechanic and continue after walls


i need revert the pr and all issuse are solved
